app/internal/import_file.py

```diff
@@ -0,0 +1,193 @@
+    return file.lower().endswith(extension)
+
+
+def is_file_exist(file: str) -> bool:
+    return Path(file).is_file()
+
+
+def is_date_in_range(date: Union[str, datetime.datetime],
+                     valid_dates: int = VALID_YEARS) -> bool:
+    """
+    check if date is valid and in the range according to the rule we have set
+    """
+    now_year = datetime.datetime.now().year
+    if isinstance(date, str):
+        try:
+            check_date = datetime.datetime.strptime(date, "%m-%d-%Y")
+        except ValueError:
+            return False
+    else:
+        check_date = date
+    return now_year - valid_dates < check_date.year < now_year + valid_dates
+
+
+def is_event_text_valid(row: str) -> bool:
+    """Check if the row contains valid data"""
+    get_values = EVENT_PATTERN.findall(row)
+    if get_values and len(get_values[0]) == NUM_OF_VALUES:
+        return True
+    return False
+
+
+def is_file_valid_to_import(file: str) -> bool:
+    """
+    checking before importing that the file exist, the file extension and
+    the size meet the rules we have set.
+    """
+    return is_file_exist(file) and is_file_extension_valid(file) and \
+        is_file_size_valid(file)
+
+
+def is_file_valid_to_save_to_database(events: List[Dict[str, Union[str, Any]]],
+                                      max_event_start_date: int
+                                      = MAX_EVENTS_START_DATE) -> bool:
+    """
+    checking after importing that there is no larger quantity of events
+    with the same date according to the rule we have set.
+    """
+    same_date_counter = 1
+    date_n_count = defaultdict(int)
+    for event in events:
+        if event["S_Date"] in date_n_count:
+            date_n_count[event["S_Date"]] += 1
+            if date_n_count[event["S_Date"]] > same_date_counter:
+                same_date_counter = date_n_count[event["S_Date"]]
+        else:
+            date_n_count[event["S_Date"]] = 1
+    return same_date_counter <= max_event_start_date
+
+
+def import_txt_file(txt_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    with open(txt_file, "r") as text:
+        events = text.readlines()
+    for event in events:
+        if not is_event_text_valid(event):
+            return list()
+        head, content, start_date, end_date = event.split(", ")
+        if not is_date_in_range(start_date) or \
+           not is_date_in_range(end_date.replace("\n", "")):
+            return list()
+        start_date = datetime.datetime.strptime(start_date, "%m-%d-%Y")
+        end_date = datetime.datetime.strptime(end_date.replace("\n", ""),
+                                              "%m-%d-%Y")
+        calendar_content.append({"Head": head,
+                                 "Content": content,
+                                 "S_Date": start_date,
+                                 "E_Date": end_date})
+    return calendar_content
+
+
+def open_ics(ics_file: str):
```

Comment:
What is the return type?

Reply:
Done. return type Calendar (from icalendar library)